src/main/MainController.java

```diff
@@ -0,0 +1,295 @@
+            System.err.println("Error filling in database from tableview");
+
+            ex.printStackTrace(System.err);
+        }
+    }
+
+    private List<Product> generateInitialProducts() throws SQLException, ClassNotFoundException {
+        return model.ProductDAO.getInitialProducts();
+    }
+
+    @FXML
+    private void search(ActionEvent actionEvent) throws ClassNotFoundException, SQLException {
+        productTable.getItems().setAll(model.ProductDAO.search(search.getText()));
+    }
+
+    @FXML
+    private void refresh(ActionEvent actionEvent) throws ClassNotFoundException, SQLException {
+        initialize();
+    }
+
+    @FXML
+    private void Exit(ActionEvent actionEvent) throws ClassNotFoundException, SQLException {
+        Platform.exit();
+
+    }
+
+    @FXML
+    private void openCSV(ActionEvent actionEvent) throws ClassNotFoundException, SQLException {
+        File file = fileChooser.showOpenDialog(pane.getScene().getWindow());
+        if (file != null && file.isFile() && file.getName().endsWith(".csv")) {
+            try (BufferedReader reader = Files.newBufferedReader(file.toPath())) {
+                String line = reader.readLine();
+                int lineCounter = 0;
+                while (line != null) {
+                    if (lineCounter == 0) {
+                        line = reader.readLine();
+                    } else {
+                        String[] productLine = line.split(";");
+                        List<String> productLineList = new LinkedList<String>(Arrays.asList(productLine));
+                        if (productLineList.size() > 4) { // maak hier pop-up van!
+                            System.out.println("Error in the CSV file!");
+                            break;
+                        }
+                        switch (productLineList.size()) {
+                            case 4:
+                                Product product = createProduct(productLineList);
+                                model.ProductDAO.addProduct(product);
+                                line = reader.readLine();
+                                break;
+                            case 3:
+                                productLineList.add("");
+                                Product product1 = createProduct(productLineList);
+                                model.ProductDAO.addProduct(product1);
+                                line = reader.readLine();
+                                break;
+                            case 2:
+                                productLineList.add("");
+                                productLineList.add("");
+                                Product product2 = createProduct(productLineList);
+                                model.ProductDAO.addProduct(product2);
+                                line = reader.readLine();
+                                break;
+                            case 1:
+                                productLineList.add("");
```

Comment:
Kunnen we dit niet inn een while loop doen ipv al deze cases?
Zolang productLineList < 4 voegen we "" toe